diff -urN redis_sessions.old/session.py redis_sessions/session.py
--- redis_sessions.old/session.py	2019-01-09 18:41:18.530462000 -0500
+++ redis_sessions/session.py	2019-01-09 18:41:39.923294153 -0500
@@ -30,6 +30,8 @@
 
             if settings.SESSION_REDIS_URL is not None:
                 self.connection_type = 'redis_url'
+            elif settings.SESSION_REDIS_CLUSTER:
+                self.connection_type = 'redis_cluster'
             elif settings.SESSION_REDIS_HOST is not None:
                 self.connection_type = 'redis_host'
             elif settings.SESSION_REDIS_UNIX_DOMAIN_SOCKET_PATH is not None:
@@ -85,6 +87,15 @@
                 db=settings.SESSION_REDIS_DB,
                 password=settings.SESSION_REDIS_PASSWORD
             )
+        elif self.connection_type == 'redis_cluster':
+            from rediscluster import RedisCluster
+            self.__redis[self.connection_key] = RedisCluster(
+                host=settings.SESSION_REDIS_HOST,
+                port=settings.SESSION_REDIS_PORT,
+                socket_timeout=settings.SESSION_REDIS_SOCKET_TIMEOUT,
+                nodemanager_follow_cluster=True,
+                skip_full_coverage_check=True
+            )
         elif self.connection_type == 'redis_unix_url':
             self.__redis[self.connection_key] = redis.StrictRedis(
                 unix_socket_path=settings.SESSION_REDIS_UNIX_DOMAIN_SOCKET_PATH,
diff -urN redis_sessions.old/settings.py redis_sessions/settings.py
--- redis_sessions.old/settings.py	2019-01-09 18:41:18.531461000 -0500
+++ redis_sessions/settings.py	2019-01-09 18:41:28.734398208 -0500
@@ -12,6 +12,7 @@
 SESSION_REDIS_PASSWORD = SESSION_REDIS.get('password', None)
 SESSION_REDIS_UNIX_DOMAIN_SOCKET_PATH = SESSION_REDIS.get('unix_domain_socket_path', None)
 SESSION_REDIS_URL = SESSION_REDIS.get('url', None)
+SESSION_REDIS_CLUSTER = SESSION_REDIS.get('cluster', None)
 
 
 """
@@ -41,4 +42,4 @@
 
 # should be on the format [(host, port), (host, port), (host, port)]
 SESSION_REDIS_SENTINEL_LIST = getattr(settings, 'SESSION_REDIS_SENTINEL_LIST', None)
-SESSION_REDIS_SENTINEL_MASTER_ALIAS = getattr(settings, 'SESSION_REDIS_SENTINEL_MASTER_ALIAS', None)
\ No newline at end of file
+SESSION_REDIS_SENTINEL_MASTER_ALIAS = getattr(settings, 'SESSION_REDIS_SENTINEL_MASTER_ALIAS', None)
